CMakeLists.txt

```diff
@@ -0,0 +1,236 @@
+
+cmake_minimum_required(VERSION 3.15)
+cmake_policy(SET CMP0048 NEW) # project version
+cmake_policy(SET CMP0076 NEW) # full paths
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake_modules")
+
+########################################################################
+# Project Details
+project(FreeRTOS-Plus-TCP
+        VERSION 3.1.0
+        DESCRIPTION "FreeRTOS TCP/UDP Network Layer"
+        HOMEPAGE_URL https://freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/index.html
+        LANGUAGES C)
+
+# Do not allow in-source build.
+if( ${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR} )
+message( FATAL_ERROR "In-source build is not allowed. Please build in a separate directory, such as ${PROJECT_SOURCE_DIR}/build." )
+endif()
+
+# Options
+option(FREERTOS_PLUS_TCP_BUILD_TEST "Build the test for FreeRTOS Plus TCP" OFF)
+
+# Configuration
+# Override these at project level with:
+#   Optional: set(FREERTOS_PLUS_TCP_BUFFER_ALLOCATION "1" CACHE STRING "" FORCE)
+#   Optional: set(FREERTOS_PLUS_TCP_COMPILER "" CACHE STRING "" FORCE)
+#   Required: set(FREERTOS_PLUS_TCP_NETWORK_IF "POSIX" CACHE STRING "" FORCE)
+
+# Select the appropriate buffer allocaiton method.
+# See: https://freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/Embedded_Ethernet_Buffer_Management.html
+if (NOT FREERTOS_PLUS_TCP_BUFFER_ALLOCATION)
```

Comment:
Could each buffer allocation be exposed as a separate static lib target?

Reply:
I treated this similar to how the HEAP selection is performed in FreeRTOS-Kernel.  

Option 1: Each BufferAllocation be exposed as separate static lib target:
-  +ves Always compiled and available.
-  -ves Compiling code that the user does not use.  Additional static libraries generated.
Option 2: port specific - it is not port specific feature - a port may limit the BufferAllocation used, but they are independent configurations.

Will leave as is, unless a specific change is requested.